tpp-app/src/onboarding/PeriodStart.js

```diff
@@ -1,42 +1,188 @@
-        <SkipButton title="Skip" onPress={() => navigation.navigate(STACK_SCREENS.SYMPTOMS_CHOICES)}/>
-        <NextButton title="Next" onPress={() => 
-          {
-            PostInitialPeriodStart(new Date(2011, 11, 30));
-            navigation.navigate(STACK_SCREENS.SYMPTOMS_CHOICES);
-          }}/>
-      </TwoButtonContainer>
-    </ImageBackground>
+    <PaperProvider theme={theme}>
+      <ImageBackground source={OnboardingBackground} style={styles.container}>
+        <BackButtonContainer>
+          <BackButton title="" onPress={() => {navigation.navigate(STACK_SCREENS.PERIOD_LENGTH)}}/>
+        </BackButtonContainer>
+
+        <SafeAreaView pointerEvents="box-none" style={{ alignItems: 'center' }}>
+          <BackgroundShape style={{ top: 10 }}/>
+          <PeriodStartIcon width='130' height='130' style={{ bottom: "38%" }}/>
+          <BarIcon style={{ bottom: "30%" }}/>
+
+          <TitleText style={{ bottom: "28%" }}>
+            When did your {'\n'} period last start?
+          </TitleText>
+          <BodyText style={{ bottom: "28%" }}>
+            Record your last period or {'\n'} skip if you don’t know
+          </BodyText>
+
+          <InputContainer style={{ bottom: "25%" }}>
+            <DatePickerButton 
+              title={range.startDate ? getCustomDateString(range.startDate) : "Choose date"} 
+              onPress={() => setOpen(true)}
+              inputted={range.startDate}>
+            </DatePickerButton>
+            <CalendarIconPref/>
+          </InputContainer>
+          <DatePickerModal 
+            backgroundColor="#000000"
+            mode="range"
+            visible={open}
+            onDismiss={onDismiss}
+            startDate={range.startDate}
+            // calculates endDate if user entered periodLength; asks user to pick endDate manually otherwise
+            endDate={periodLength && range.startDate ? getEndDate() : range.endDate}
+            onConfirm={onConfirm}
+            onChange={onChange}
+            validRange={{ endDate: new Date() }}
+            saveLabel="Done" 
+            uppercase={false}
+            label="Select Start Date"
+            startLabel="From"
+            endLabel="To"
+          />
+        </SafeAreaView>
+ 
+        <TwoButtonContainer>
+          <SkipButton title="Skip" onPress={() => navigation.navigate(STACK_SCREENS.SYMPTOMS_CHOICES, {
+            periodLength: periodLength,
+            periodStart: null,
+            periodEnd: null
+          })}/>
+          <NextButton title="Next" onPress={() => 
+            {
+              if(periodLength == null) {
+                // user did not enter periodLength previously but manually selected a complete date range
+                newPeriodLength = Math.round((range.endDate.getTime() - range.startDate.getTime()) / MILLISECPERDAY);
+                PostInitialPeriodLength(newPeriodLength);
+              }
+              PostInitialPeriodStart(range.startDate, range.endDate);
```

Comment:
Something I forgot to ask on your endpoints PR, but for your endpoints, I feel that it's more stylistically consistent to write the "post", "get" etc. in all caps in your endpoint names. Not a big deal obviously, more like I noticed everyone else have been adhering to this style

Reply:
changed now - i didn't even realize thank you annie!!